apps/emqx/src/emqx_channel.erl

```diff
@@ -1453,22 +1499,36 @@ handle_call(
         conninfo = #{clientid := ClientId}
     }
 ) ->
-    %% NOTE
-    %% This is essentially part of `emqx_session_mem` logic, thus call it directly.
-    ok = emqx_session_mem:takeover(Session),
-    %% TODO: Should not drain deliver here (side effect)
-    Delivers = emqx_utils:drain_deliver(),
-    AllPendings = lists:append(Pendings, maybe_nack(Delivers)),
-    ?tp(
-        debug,
-        emqx_channel_takeover_end,
-        #{clientid => ClientId}
-    ),
-    Channel0 = maybe_publish_will_msg(takenover, Channel),
-    disconnect_and_shutdown(takenover, AllPendings, Channel0);
+    ?EXT_TRACE_WITH_PROCESS_FUN(
```

Comment:
Does the `broker_disconnect` trigger twice here? The first is triggered by sock_closed, and The second is triggered by `{takeover, end}`.